src/xeus_sqlite_interpreter.cpp

```diff
@@ -111,17 +111,12 @@ void interpreter::load_db(const std::vector<std::string> tokenized_code)
         }
         else
         {
-            std::string error = "Wasn't able to load the database correctly.";
-            m_traceback.push_back("Error: " + error);
-            publish_execution_error("Error", error, m_traceback);
-            m_traceback.clear();
+            throw std::runtime_error("Wasn't able to load the database correctly.");
         }
     }
     catch (const std::runtime_error& err)
     {
-        m_traceback.push_back("Error: " + (std::string)err.what());
-        publish_execution_error("Error", err.what(), m_traceback);
-        m_traceback.clear();
+        throw std::runtime_error("Error: " + (std::string)err.what());
```

Comment:
Actually no need to do the `try/catch` here. Just execute the code :)

So your `load_db` method can contain only:
```cpp
if (tokenized_code.back().find("rw") != std::string::npos)
{
    m_bd_is_loaded = true;
    m_db = std::make_unique<SQLite::Database>(m_db_path,
                SQLite::OPEN_READWRITE);
}
else if (tokenized_code.back().find("r") != std::string::npos)
{
    m_bd_is_loaded = true;
    m_db = std::make_unique<SQLite::Database>(m_db_path,
                SQLite::OPEN_READONLY);
}
//Opening as read and write because mode is unspecified
else if (tokenized_code.size() < 4)
{
    m_bd_is_loaded = true;
    m_db = std::make_unique<SQLite::Database>(m_db_path,
                SQLite::OPEN_READWRITE);
}
else
{
    throw std::runtime_error("Wasn't able to load the database correctly.");
}
```